patterns/behavioral/command.py

```diff
@@ -14,33 +20,65 @@
 https://docs.djangoproject.com/en/2.1/ref/request-response/#httprequest-objects
 """
 
+import os
+
 
-class Command:
+class HideFileCommand:
     """
-    The interface that commands implement. This is used to abstract invoker from
-    the command is going to handle the job.
+    A command to hide a file given its name
     """
 
-    def execute(self):
-        raise NotImplementedError()
+    def __init__(self):
+        # an array of files hidden, to undo them as needed
+        self._hidden_files = []
 
+    def execute(self, filename):
+        if os.path.isfile(filename):
+            print(f'hiding {filename}')
 
-class MakeBoldCommand(Command):
-    """
-    A simple command to bold a text.
-    """
+            os.rename(filename, f'.{filename}')
+            self._hidden_files.append(filename)
+        else:
+            print(f'{filename} dose not exists to hide')
+
+    def undo(self):
+        if len(self._hidden_files) > 0:
+            filename = self._hidden_files.pop()
 
-    def execute(self):
-        print('I am making it bold.')
+            print(f'un-hiding {filename}')
 
+            os.rename(f'.{filename}', filename)
 
-class MakeItalicCommand(Command):
+
+class DeleteFileCommand:
     """
-    A simple command to italic a text.
+    A command to delete a file given its name
     """
 
-    def execute(self):
-        print('I am making it italic.')
+    def __init__(self):
+        # an array of deleted files, to undo them as needed
+        self._deleted_files = []
+
+        # create a directory to store deleted files
+        if not os.path.exists('bin'):
```

Comment:
Please choose something else (maybe "trash)" since `bin` is a common name for storing executable files and might cause confusion